### PR TITLE
Replace double URL escaping with concats as the base URLs are already…

### PR DESCRIPTION
**Pull request by seanson**
… escaped. Fix for #402

### CODE REVIEW

File: notebook/static/services/sessions/session.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ define([
 
         this.base_url = options.base_url;
         this.ws_url = options.ws_url;
-        this.session_service_url = utils.url_join_encode(this.base_url, 'api/sessions');
+        this.session_service_url = this.base_url + '/api/sessions');
```
Comment (minrk) [inline notebook/static/services/sessions/session.js]: closing parenthesis should be removed

File: notebook/static/services/sessions/session.js [hunk 2]
```diff
@@ -259,7 +259,7 @@ define([
     Session.prototype._update_model = function (data) {
         if (data && data.id) {
             this.id = data.id;
-            this.session_url = utils.url_join_encode(this.session_service_url, this.id);
+            this.session_url = this.session_service_url + '/' + this.id);
```
Comment (minrk) [inline notebook/static/services/sessions/session.js]: same here

**Comment by minrk** [comment]
:+1: to the fix once the syntax errors are resolved.

**Comment by seanson** [comment]
Apologies for the multiple amends, I had an additional / in the concat that caused a failure in the build tests / running locally. Last commit passes testing.

File: notebook/static/services/sessions/session.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ define([
 
         this.base_url = options.base_url;
         this.ws_url = options.ws_url;
-        this.session_service_url = utils.url_join_encode(this.base_url, 'api/sessions');
+        this.session_service_url = this.base_url + 'api/sessions';
```
Comment (minrk) [inline notebook/static/services/sessions/session.js]: to ensure slashes are correct, you can use `utils.url_join` instead of `url_join_encode`.

**Comment by minrk** [comment]
@seanson if you want to squash your changes into one commit, you can use `git rebase -i master` and `squash` (or `fix`) all your commits together, e.g.

```git
pick 6ad8ea2 Replace...
s fa02d0e Replace...
s 88359c4 Replace...

# Rebase 159b470..88359c4 onto 159b470 (3 command(s))
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

You can also you `git commit --amend` to amend your latest commit when you are adding fixes. If you rewrite history, you will have to `git push -f` to confirm that the remote branch should be rewritten instead of added to.

**Comment by seanson** [comment]
Thanks for the help! I'm still a bit new to the git workflow so the advice is definitely welcome! 

There appears to have been some sort of Travis CI failure for a pip install among all the tests but they're passing locally.

File: notebook/static/services/sessions/session.js [hunk 4]
```diff
@@ -259,7 +259,15 @@ define([
     Session.prototype._update_model = function (data) {
         if (data && data.id) {
             this.id = data.id;
-            this.session_url = utils.url_join_encode(this.session_service_url, this.id);
+<<<<<<< HEAD
+<<<<<<< HEAD
```
Comment (minrk) [inline notebook/static/services/sessions/session.js]: Looks like some conflict markers have been committed here. You want to be careful about resolving conflicts before merging.

**Comment by minrk** [comment]
@seanson do you need a hand finishing this one up? I can do the rebase & merge, if you would prefer.

**Comment by seanson** [comment]
Apologies for the delays, I will likely be able to attend this later tonight (GMT+10). I may close and re-open the PR with a proper implementation with the url join from utils as well.

**Comment by Carreau** [comment]
> Apologies for the delays, I will likely be able to attend this later tonight (GMT+10). I may close and re-open the PR with a proper implementation with the url join from utils as well.

No hurry. 

You can just push on the same branch and this PR will update. 

Thanks.

**Comment by minrk** [comment]
@seanson plans to finish this up? I'm happy to take it over the finish line if you are busy.

**Comment by seanson** [comment]
Thanks for the patience and the help with the git amending. I've redone the commit with the built-in utils calls and looks like tests are good!

**Comment by minrk** [comment]
@seanson huzzah!

**Pull request merged**